File: https:/github.com/rajesh-kayal-dev/Leetcode.git/344-ReverseString/344-ReverseString.java
// Last updated: 6/15/2026, 10:00:40 AM
class Solution {
    public void reverseString(char[] s) {
        // int n =s.length -1;
        // for(int i =0;i<=n;i++){
        //     char temp = s[i];
        //     s[i]=s[n];
        //     s[n]=temp;
        //     n--;
        // }



        int start =0;
        int end = s.length -1;
        while( start <= end){
            char temp = s[start];
            s[start]=s[end];
            s[end]=temp;
            start ++;
            end --;
        }
    }
}

File: https:/github.com/rajesh-kayal-dev/Leetcode.git/344-ReverseString/344-ReverseString.cpp
// Last updated: 6/15/2026, 10:00:47 AM
class Solution {
public:
    void reverseString(vector<char>& s) {
        int start = 0;
        int end = s.size()-1;
        while(start < end)
        {
            swap(s[start],s[end]);
            start++;
            end--;
        }
        
    }
};